src/functionalTest/java/uk/gov/hmcts/reform/professionalapi/idam/IdamOpenIdClient.java

```diff
@@ -66,21 +66,31 @@ public IdamOpenIdClient(TestConfigProperties testConfig) {
         String serializedUser = gson.toJson(user);
         //required logs for debug
         //log.info("serializedUser: " + serializedUser);
-        Response createdUserResponse = RestAssured
-            .given()
-            .relaxedHTTPSValidation()
-            .baseUri(testConfig.getIdamApiUrl())
-            .header(CONTENT_TYPE, APPLICATION_JSON_VALUE)
-            .body(serializedUser)
-            .post("/testing-support/accounts")
-            .andReturn();
 
+        Response createdUserResponse = null;
+
+        for (int i = 0; i < 5; i++) {
+            log.info("SIDAM createUser retry attempt : " + i + 1);
+            createdUserResponse = RestAssured
+                    .given()
+                    .relaxedHTTPSValidation()
+                    .baseUri(testConfig.getIdamApiUrl())
+                    .header(CONTENT_TYPE, APPLICATION_JSON_VALUE)
+                    .body(serializedUser)
+                    .post("/testing-support/accounts")
+                    .andReturn();
+            if (createdUserResponse.getStatusCode() == 504) {
```

Comment:
Can we reverse the condition, i.e. if status is !201?

Reply:
We would only want to retry in 504 scenario I am assuming - if the response was 400, we would want to know rather than sending off a further 4 requests?

Reply:
Makes sense
thanks